osf-cli/README.md

```diff
@@ -0,0 +1,83 @@
+
+##How It Works
+The CLI uses functionality in the `osf-packager` module and the OSF Java client in the `osf-client` package to attach to the OSF REST APIs (both v1 and v2) on a running OSF instance specified by a configuration file whose location must be supplied by the user. Additional parameters will need to be supplied as indicated below. The packager code leverages a workflow from the Dataconservancy Package Tool GUI to construct the package once the content from the target registration has been retrieved. The package is then saved to a location specified by the user as a command line option.
+
+
+##Command Line Usage
+The CLI takes a single argument which is a URL pointing to the registration whcih is to be packaged (for example, `http://192.168.99.100:8000/v2/registrations/vacbu/`
+This is preceded by a list of command line options as follows:
+
+```
+-c (-configuration, --configuration)  FILE   : path to the OSF Java client configuration
+-h (-help, --help)                           : print help message
+-m (-metadata, --metadata) FILE              : the path to the metadata properties file for additional bag metadata
+-n (-name, --name) VAL                       : the name for the bag
+-o (-output, --output) FILE                  : path to the directory where the package will be written
+-v (-version, --version)                     : print version information
+```
+The `-c, -m, -n` and `-o` options are required. The OSF Java client must be configured so that it knows which running OSF instance to attach to. Additional metadata is specified in a bag metadata properties file, and is used to describe the bag in accordance with the Data Conservancy BagIt Profile. The package name is used to both name the package and the bag. Finally, the output location will tell the CLI where to write the package.
+
+###OSF Java Client Configuration
+Configuration mut be supplied for both the v1 and v2 endpoints, since both are needed to build the package.  An example is below:
+```
+{
+  "osf": {
+    "v2": {
+      "host": "192.168.99.100",
+      "port": "8000",
+      "basePath": "/v2/",
+      "authHeader": "Basic ZW1ldHNnZXJAZ21haWwuY29tOmZvb2JhcmJheg==",
+      "scheme": "http"
+    }
+  },
+  "wb": {
+    "v1": {
+      "host": "192.168.99.100",
+      "port": "7777",
+      "basePath": "/v1/",
+      "scheme": "http"
+    }
+  }
+}
+```
+The absolute path to the file containing this configuration is what must be supplied as the value for the `-c` option.
+
+###Bag Metadata File
+The `-m` option also takes an absolute path as its value. This file contains bag metadata which will conforn to the Data Conservancy BagIt Profile specification. Metadata entries will consist of key = value pairs, one per line. The values may be lists where appropriate, expressed as comma separated strings. If the value of Package-Name is specified in this file, it will be superseded by the value specified for the `-n` option below. 
```

Comment:
To be spec compliant (PTG compatibility aside for the moment) we require two properties in `bag-info.txt`:
* `BagIt-Profile-Identifier:  http://dataconservancy.org/formats/data-conservancy-pkg-1.0`
* `Resource-Manifest: bag://<bag name>/path/to/ore.rem`

This CLI should be responsible for adding those to properties because 1) the user isn't responsible for enforcing a Bag profile, 2) nor are they responsible for creating the resource manifest or placing it in the package.

So as I understand it, this command line flag would be an _optional_ way for a user to supply additional metadata they would want to appear in `bag-info.txt`.  Given the above, is that the way you see it as well?

If you agree, then what form would the supplied file take?  For example, a user could just point to an existing `bag-info.txt` file.  The CLI will copy all of the metadata from the user-supplied file, and overwrite any values for `BagIt-Profile-Identifier` and `Resource-Manifest` that appeared in the user-supplied `bag-info.txt`.  The other metadata that appeared in the user-supplied file would be copied verbatim.

Regarding `Package-Name` and compatibility with the PTG, IMO I don't think this is a goal of the CLI.  That is, the CLI doesn't need to produce a `bag-info.txt` file that contains or uses the same bag metadata that the PTG uses.  In the future, the PTG may be updated to open _any_ Bag compliant with the BagIt [specification](https://tools.ietf.org/html/draft-kunze-bagit-11) (not our specification but the canonical one), and so it would be able to open Bags produced by this CLI with that additional functionality.  In this regard, I don't think that the `Package-Name` metadata keyword deserves any special consideration.